### DISCUSSION TITLE
Clarify how to specify a commercial license

### DISCUSSION THREAD

**Discussion by collimarco** [ruby/rubygems#5254] [Q&A] [answered]
We have a gem distributed under a commercial license (Cuber License).

Now we have specified the license in a file called `LICENSE` in the root of the Git repository.

However when I run `gem build` I get a warning, complaining that `license` is not present in the `gemspec`.

I see a lot of confusion here, in case of a commercial license (EULA)...

1. Should I keep the `license` field empty?
2. Should I use `Nonstandard` for the `license` field?
3. Should I use `Cuber License` for the `license` field?
4. Should I use `LicenseRef-LICENSE` for the `license` field?

Every part of your documentation seems to suggest something different and a clarification would be useful for everyone.

**Comment by simi** [comment]
I'm not SPDX expert, but just follow SPDX specification - https://spdx.github.io/spdx-spec/.

**Comment by collimarco** [comment]
@simi Which, in practical terms, is case 1, 2, 3, 4 or something else? Can you provide a concrete solution for my scenario above (which is a common case where you have a commercial license)?

**Comment by simi** [comment]
I have no idea, please consult with SPDX specification and experts. Anyway I don't see anything wrong here with RubyGems.

**Comment by simi** [comment]
_moved to discussion_

**Reply by collimarco** [reply]
I don't know why you moved this to a discussion, when it's actually an issue.

RubyGems should at least clarify the documentation.

**The documentation must specify clearly a single approach for commercial licenses.**

Please reopen the issue until this is fixed. Thanks

**Comment by simi** [comment]
According to https://github.com/rubygems/rubygems/pull/5013, I would go with `4.`.

**Reply by collimarco** [reply]
If this is the correct case, then you must update the output of `gem build`, which suggests something different.

Please reopen the issue until this is clarified

**Reply by deivid-rodriguez** [reply]
I think it's fine to keep this a discussion until clarified. Once clarified, you can directly create a pull request to improve the documentation and/or `gem build` output.

I agree that 4. seems like the best option for this case.

**Reply by simi** [reply]
@collimarco you need RubyGems v3.3.0+ to get support for `LicenseRef-*`. Can you check `LicenseRef-*` causes trouble even using RubyGems 3.3.0 (or later)?

**Reply by collimarco** [reply]
Following your advice, in the previous example I would use `LicenseRef-LICENSE` (since the file, like in most repos, is called simply `LICENSE`). Someone who reads the `gemspec` may find it strange to read `LicenseRef-LICENSE`... What does it mean? It's not very intuitive. I am not saying it's wrong, just not very meaningful and possibly misleading. Do you have any idea where this convention comes from? Can we find a better alternative (useful for all commercial or private Ruby gems)?

**Reply by simi** [reply]
Is is coming from https://spdx.github.io/spdx-spec/ (Annex D.1) as I mentioned before. Gemspec is not meant to be used directly, anyway most of the columns are documented at proper place - https://guides.rubygems.org/specification-reference/#license=. It would be nice if you please try to spend your own time looking for the answers first. I tried to point to to proper places already few times.

**Reply by collimarco** [reply]
@simi I already read that documentation before opening the issue, as you can clearly see from my original post. In any case it seems that you just want to close this discussion and your attitude, from the beginning, is definitely hostile towards new contributions and improvements, so I won't post anything else.  It's sad, because this issue is relevant for everyone (clarifying how commercial licenses are specified allows a better use of automation tools for license checks and avoids ambiguity).

**Reply by simi** [reply, marked as answer]
Even I don't like your tone of writing, I have answered all your questions.

1. in your case use `LicenseRef-LICENSE`
2. for another valid values consult SPDX docs or consult with SPDX people
3. documentation for gemspec `license` attribute is present at https://guides.rubygems.org/specification-reference/#license=

If you have any idea to improve, feel free to share.

**Comment by deivid-rodriguez** [comment]
Closing as answered.
